src/storage/src/hummock/store/version.rs

```diff
@@ -283,27 +465,90 @@ impl HummockReadVersion {
             }
 
             VersionUpdate::CommittedSnapshot(committed_version) => {
-                let max_committed_epoch = committed_version.max_committed_epoch();
-                self.committed = committed_version;
+                let cleaned_ssts = self.clean_staging_ssts(committed_version);
 
-                {
-                    // TODO: remove it when support update staging local_sst
-                    self.staging
-                        .imm
-                        .retain(|imm| imm.epoch() > max_committed_epoch);
-                    self.staging.sst.retain(|sst| {
-                        sst.epochs.first().expect("epochs not empty") > &max_committed_epoch
-                    });
-
-                    // check epochs.last() > MCE
-                    assert!(self.staging.sst.iter().all(|sst| {
-                        sst.epochs.last().expect("epochs not empty") > &max_committed_epoch
-                    }));
+                if let Some(committed_index) = self.committed_index.as_deref() {
+                    self.committed_index = Some(Arc::new(
+                        committed_index.apply(self.committed(), cleaned_ssts),
+                    ));
                 }
             }
         }
     }
 
+    fn clean_staging_ssts(
```

Comment:
The logic in this function is unclear. It is better not mixing the logic of clearing staging ssts and calculating the diff between cleared staing ssts & pruned version together.

What I think the function does by looking at the function name and the parameters:
- clear the stale staging ssts by checking committed_version 
- returns the cleared staging ssts.

What this function actually does:
- clear the stale staging ssts by checking committed_version
- calculate the diff between staging ssts and pruned version
- return the diff

Reply:
> The logic in this function is unclear. It is better not mixing the logic of clearing staging ssts and calculating the diff between cleared staing ssts & pruned version together.
> 
> What I think the function does by looking at the function name and the parameters:
> 
> * clear the stale staging ssts by checking committed_version
> * returns the cleared staging ssts.
> 
> What this function actually does:
> 
> * clear the stale staging ssts by checking committed_version
> * calculate the diff between staging ssts and pruned version
> * return the diff

OK. I can use `assert` to make them consistent.

Reply:
> The logic in this function is unclear. It is better not mixing the logic of clearing staging ssts and calculating the diff between cleared staing ssts & pruned version together.
> 
> What I think the function does by looking at the function name and the parameters:
> 
> * clear the stale staging ssts by checking committed_version
> * returns the cleared staging ssts.
> 
> What this function actually does:
> 
> * clear the stale staging ssts by checking committed_version
> * calculate the diff between staging ssts and pruned version
> * return the diff

fixed